### ISSUE TITLE
s3.getSignedUrl() keeps returning undefined

### ISSUE THREAD

**Issue by marcosCapistrano** [async-labs/saas#166]
Not sure if im doing something wrong, tried it a bunch of times and it keeps returning undefined here:

  return new Promise((resolve, reject) => {
    s3.getSignedUrl('putObject', params, (err, data) => {
      const parsedUrl = url.parse(data); <<<<<<<<<<< HERE "data" is undefined

      const returnData = {
        signedRequest: data,
        url: `${parsedUrl.protocol}//${parsedUrl.hostname}${parsedUrl.pathname}`,
      };

      if (err) {
        console.error(err);
        reject(err);
      } else {
        resolve(returnData);
      }
    });
  });
  
  Files from 5-begin can be used for testing

**Comment by klyburke** [comment]
@marcosCapistrano Thanks for reporting. I will run 5-begin and see if I can reproduce the error.

**Comment by klyburke** [comment]
@marcosCapistrano I ran the `5-begin` code and did not see that `data` in that line of code is undefined.

To confirm that `data` is not undefined here, I went to the `aws-s3.ts` file of `5-begin` and added `console.log(data)` like so:

![Screenshot from 2021-09-14 14-10-44](https://user-images.githubusercontent.com/26158226/133335950-f61ae029-68ba-4bd3-aa4d-de1a523986e0.png)

I think started the app, went to `Your Settings` page and uploaded a new avatar. After that, my terminal of VS editor outputs `data`. Hence, it is not undefined.

![Group 141](https://user-images.githubusercontent.com/26158226/133349853-07cc2612-9aca-4444-9118-23adccacd8e5.png)

---

I would look at your AWS S3 bucket. Under `Permissions`, make sure you deselected `Block all public access`:

![Screenshot from 2021-09-14 14-28-12](https://user-images.githubusercontent.com/26158226/133336572-2dbaf0c6-e9ac-4bea-aa0b-04bb1170dfb8.png)

Also under `Permissions`, make sure your CORS policy is correct:

![Screenshot from 2021-09-14 14-30-43](https://user-images.githubusercontent.com/26158226/133336766-81bf8bef-bb83-4264-b50b-40b094f9592c.png)

Finally, check that you have all necessary environmental variables saved in your appropriate `.env` files.

---

Let me know if you still have trouble. Feel free to create a public repo with your code, and I will test.
